methods/did:key.md

```diff
@@ -0,0 +1,47 @@
+# did:key
+
+The [Key DID method](https://w3c-ccg.github.io/did-method-key) encoding leverages the existing multicodes used by the Key DID method.
+
+## Method multicodec
+
+| Code   | Method identifier length (bytes)                             |
+| ------ | ------------------------------------------------------------ |
+| 0xe7   | 33                                                           |
+| 0xea   | ?? (not part of the Key DID spec)                            |
+| 0xeb   | 96                                                           |
+| 0xec   | 32                                                           |
+| 0xed   | 32                                                           |
+| 0x1200 | 33                                                           |
+| 0x1201 | 49                                                           |
+| 0x1202 | 67                                                           |
+| 0x1205 | 270 or 526 (see [RSA Public Key representation](https://w3c-ccg.github.io/did-method-key/#rsa-repr) for how to distinguish these two cases) |
+
+## Format
+
+```
+<multidid-code><method-code><method-bytes><url-length><url-bytes>
+```
+
+Where:
+
+* `multidid-code` - a varint encoding of `0xd1d`
+* `method-code` - a varint encoding of one of the codes in the table above
+* `method-bytes` - MUST contain exactly the number of bytes specified byte the method code
+* `url-length` - a varint describing the length of the `url-bytes` parameter
+* `url-bytes` - a UTF-8 encoded string representing the [DID URL parameters](https://www.w3.org/TR/did-core/#did-url-syntax)
+
+## Examples
+
+Displayed using base 16.
+
+```
+// did:key:z6MkiTBz1ymuepAQ4HEHYSF1H8quG5GLVVQR3djdX3mDooWp
+0x9d1aed01ed013b6a27bcceb6a42d62a3a8d02a6f0d73653215771de243a63ac048a18b59da29
+
+// did:key:z6MkiTBz1ymuepAQ4HEHYSF1H8quG5GLVVQR3djdX3mDooWp#z6MkiTBz1ymuepAQ4HEHYSF1H8quG5GLVVQR3djdX3mDooWp
+0x9d1aed01ed013b6a27bcceb6a42d62a3a8d02a6f0d73653215771de243a63ac048a18b59da2931237a364d6b6954427a31796d75657041513448454859534631483871754735474c5656515233646a6458336d446f6f5770
+
+// did:key:zQ3shokFTS3brHcDQrn82RUDfCZESWL1ZdCEJwekUDPQiYBme
+0x9d1ae70103874c15c7fda20e539c6e5ba573c139884c351188799f5458b4b41f7924f235cd
+```
```

Comment:
```sh
npm i multibase
```
```js
const multibase = require('multibase')
```

`<multidid-code><method-code><method-bytes><url-length><url-bytes>`

`// did:key:z6MkiTBz1ymuepAQ4HEHYSF1H8quG5GLVVQR3djdX3mDooWp`
`0x9d1a ed01 ed01 3b6a27bcceb6a42d62a3a8d02a6f0d73653215771de243a63ac048a18b59da29`
```js
0x9d1a  // multidid-code 0x0d1d
0xed01  // method-code 0xed ed25519-pub
0x3b6a27bcceb6a42d62a3a8d02a6f0d73653215771de243a63ac048a18b59da29  // method-bytes
"" // url-length
"" // url-bytes
```

```js
varint.decode(multibase.decode("z6MkiTBz1ymuepAQ4HEHYSF1H8quG5GLVVQR3djdX3mDooWp")).toString(16)  // 'ed' ed25519-pub
varint.decode.bytes // 2 vint is first 2 bytes
Array.from(multibase.decode("z6MkiTBz1ymuepAQ4HEHYSF1H8quG5GLVVQR3djdX3mDooWp")).slice(2).map(i => i.toString(16)).padStart(2, '0').join("") // '3b6a27bcceb6a42d62a3a8d02a6f0d73653215771de243a63ac048a18b59da29'
```

`// did:key:z6MkiTBz1ymuepAQ4HEHYSF1H8quG5GLVVQR3djdX3mDooWp#z6MkiTBz1ymuepAQ4HEHYSF1H8quG5GLVVQR3djdX3mDooWp`
`0x9d1a ed01 ed01 (I think this second ed01 a bug) 3b6a27bcceb6a42d62a3a8d02a6f0d73653215771de243a63ac048a18b59da2931237a364d6b6954427a31796d75657041513448454859534631483871754735474c5656515233646a6458336d446f6f5770`

```js
0x9d1a  // multidid-code 0x0d1d
0xed01  // method-code 0xed ed25519-pub
0x3b6a27bcceb6a42d62a3a8d02a6f0d73653215771de243a63ac048a18b59da29  // method-bytes
0x31 // url-length
0x237a364d6b6954427a31796d75657041513448454859534631483871754735474c5656515233646a6458336d446f6f5770 // url-bytes
```

```js
Array.from(multibase.decode("z6MkiTBz1ymuepAQ4HEHYSF1H8quG5GLVVQR3djdX3mDooWp")).slice(2).map(i => i.toString(16).padStart(2, '0')).join("")  // '3b6a27bcceb6a42d62a3a8d02a6f0d73653215771de243a63ac048a18b59da29'
"#z6MkiTBz1ymuepAQ4HEHYSF1H8quG5GLVVQR3djdX3mDooWp".length  // 49

varint.encode(49).map((x) => x.toString(16).padStart(2, '0')).join("")  // '31'

Array.from("#z6MkiTBz1ymuepAQ4HEHYSF1H8quG5GLVVQR3djdX3mDooWp", (x) => x.charCodeAt(0).toString(16).padStart(2, '0')).join("") 
 // '237a364d6b6954427a31796d75657041513448454859534631483871754735474c5656515233646a6458336d446f6f5770'

```

`// did:key:zQ3shokFTS3brHcDQrn82RUDfCZESWL1ZdCEJwekUDPQiYBme`
`0x9d1a e701 03874c15c7fda20e539c6e5ba573c139884c351188799f5458b4b41f7924f235cd`

```js
0x9d1a  // multidid-code 0x0d1d
0xe701  // method-code 0xed ed25519-pub
0x03874c15c7fda20e539c6e5ba573c139884c351188799f5458b4b41f7924f235cd  // method-bytes
"" // url-length
"" // url-bytes
```

```js
varint.decode(multibase.decode("zQ3shokFTS3brHcDQrn82RUDfCZESWL1ZdCEJwekUDPQiYBme")).toString(16)  // 'e7' secp256k1-pub
varint.decode.bytes  // 2 vint is first 2 bytes

Array.from(multibase.decode("zQ3shokFTS3brHcDQrn82RUDfCZESWL1ZdCEJwekUDPQiYBme")).slice(2).map(i => i.toString(16).padStart(2, '0')).join("")  // 
'03874c15c7fda20e539c6e5ba573c139884c351188799f5458b4b41f7924f235cd'
```

Reply:
Yep, that looks like a bug to me @AaronGoldman. Could you open a PR to fix it?